spring-cloud-data-rest-resource/src/main/java/org/springframework/cloud/data/rest/resource/ModuleRegistrationResource.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.rest.resource;
+
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.ResourceSupport;
+
+/**
+ * Rest resource for a module registration.
+ *
+ * @author Glenn Renfro
+ * @author Mark Fisher
+ * @author Patrick Peralta
+ */
+public class ModuleRegistrationResource extends ResourceSupport {
+
+	/**
+	 * Module name.
+	 */
+	private String name;
+
+	/**
+	 * Module type.
+	 */
+	private String type;
+
+	/**
+	 * Maven coordinates for module artifact. String is in standard format
+	 * such as {@code groupId:artifactId:version}.
+	 */
+	private String coordinates;
+
+	/**
+	 * True if this is a composed module.
+	 */
+	private boolean composed;
+
+	/**
+	 * Default constructor for serialization frameworks.
+	 */
+	protected ModuleRegistrationResource() {
+	}
+
+	/**
+	 * Construct a {@code ModuleRegistrationResource}.
+	 *
+	 * @param name module name
+	 * @param type module type
+	 * @param coordinates coordinates for module artifact
+	 * @param composed true if this is a composed module
```

Comment:
If we add back `module compose`, I think it will merely be a way to save a shortcut, e.g.:

```
module compose  loudhttp --definition "http | transform --expression=payload.toUpperCase()"
```

so that streams can be defined with that:

```
stream create example --definition "loudhttp | log"
```

We need to decide where we handle the expansion of the composed module, but I think we should do that expansion prior to looking up registry entries. In other words, the above stream definition would still lead to a lookup of 3 modules: http, filter, log.